cg/meta/store/mip_rna.py

```diff
@@ -1,102 +1,176 @@
-            'path': sample_data['mark_duplicates'],
-            'tags': ['mark-duplicates', sample_data['id'], 'rd-rna'],
-            'archive': False,
-        })
-        data.append({
-            'path': sample_data['salmon_quant'],
-            'tags': ['salmon-quant', sample_data['id'], 'rd-rna'],
-            'archive': False,
-        })
-        data.append({
-            'path': sample_data['star_fusion'],
-            'tags': ['star-fusion', sample_data['id'], 'rd-rna'],
-            'archive': False,
-        })
-        data.append({
-            'path': sample_data['stringtie_ar'],
-            'tags': ['stringtie-ar', sample_data['id'], 'rd-rna'],
-            'archive': False,
-            })
+    data = [
+        {
+            "path": file["path"],
+            "tags": sorted(
+                list(
+                    set(
+                        [
+                            file["format"],
+                            file["id"],
+                            file["step"],
+                            file["tag"],
+                            "rd-rna",
+                        ]
+                    )
+                    - set([None])
+                )
+            ),
+            "archive": False,
+        }
+        for file in deliverables["files"]
+    ]
 
     return data
+
+
+def parse_config(data: dict) -> dict:
+    """Parse MIP config file.
+
+    Args:
+        data (dict): raw YAML input from MIP analysis config file
+
+    Returns:
+        dict: parsed data
+    """
+    return {
+        "email": data.get("email"),
+        "case": data["case_id"],
+        "samples": [
+            {"id": sample_id, "type": analysis_type}
+            for sample_id, analysis_type in data["analysis_type"].items()
+        ],
+        "is_dryrun": bool("dry_run_all" in data),
+        "out_dir": data["outdata_dir"],
+        "priority": data["slurm_quality_of_service"],
+        "sampleinfo_path": data["sample_info_file"],
+    }
+
+
+def parse_sampleinfo(data: dict) -> dict:
+    """Parse MIP sample info file (RNA).
+
+    Args:
+        data (dict): raw YAML input from MIP qc sample info file (RNA)
+
+    Returns:
+        dict: parsed data
+    """
+    case = data["case"]
+
+    sampleinfo_data = {
+        "date": data["analysis_date"],
+        "is_finished": data["analysisrunstatus"] == "finished",
+        "case": case,
+        "version": data["mip_version"],
+    }
+
+    return sampleinfo_data
+
+
+def get_case(bundle_obj, status):
+    """ Get a case from the status database """
+    case_obj = status.family(bundle_obj.name)
+    return case_obj
+
+
+def reset_case_action(case_obj):
+    """ Resets action on case """
+    case_obj.action = None
+
+
+def add_new_complete_analysis_record(bundle_data, case_obj, status, version_obj):
+    """Function to create and return a new analysis database record"""
+    pipeline = case_obj.links[0].sample.data_analysis
+    pipeline = pipeline if pipeline else "mip"
```

Comment:
```suggestion
    pipeline = pipeline if pipeline else "mip-rna"
```
This value should depend on the analysis used I guess and for RNA it should be something with RNA?